### PR TITLE
fix qemu x86 program memory size and align flash simulator

### PR DESCRIPTION
**Pull request by nvlsianpu**

### CODE REVIEW

**Comment by nvlsianpu** [comment]
As it is supposed to fix high priority bug I'm adding appropriate label.

**Comment by nvlsianpu** [comment]
@kapi-no I need review form you.

**Comment by nvlsianpu** [comment]
cc @ioannisg

File: drivers/flash/flash_simulator.c [hunk 1]
```diff
@@ -45,13 +46,13 @@
 	} while (0)
 
 #if (defined(CONFIG_STATS) && \
-     (FLASH_SIMULATOR_FLASH_SIZE > STATS_PAGE_COUNT_THRESHOLD))
+     (CONFIG_FLASH_SIMULATOR_STAT_PAGE_COUNT > STATS_PAGE_COUNT_THRESHOLD))
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply drivers/flash/flash_simulator.c]: Not needed. But I like defensive programming.

File: drivers/flash/Kconfig.simulator [hunk 2]
```diff
@@ -44,6 +44,19 @@ config FLASH_SIMULATOR_SIMULATE_TIMING
...
 
+config FLASH_SIMULATOR_STAT_PAGE_COUNT
+	int "Pages under statistic"
+	range 1 256
+	default 256
+	help
+	  Only up to this number of beginning pages will be tracked
+	  while catching dedicated flash operations and thresholds.
+	  This number is not automatic because implementation uses
+	  UNTIL_REPET() macro, which is limited to take explicitly
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply drivers/flash/Kconfig.simulator]: corected

File: drivers/flash/Kconfig.simulator [hunk 3]
```diff
@@ -44,6 +44,19 @@ config FLASH_SIMULATOR_SIMULATE_TIMING
...
+	  number of iterations.
+	  This is why can't to calculate pages number by preprocessor basing
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply drivers/flash/Kconfig.simulator]: fixed

**Review by kapi-no — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]
Looks ok to me

**Comment by ioannisg** [comment]
Let's do a rebase before merging, once the review round is complete.

**Comment by andrewboie** [comment]
@ioannisg GH says we can merge-as is, why do you think this needs a manual rebase?

**Review by dcpleung — APPROVED** [review]

**Comment by marc-h38** [comment]
BTW  to test against the latest master closing+reopening is enough because CI always rebases before testing. Saves confusing GitHub with one extra force push.

**Pull request merged**